modules/correlation/groupingby.c

```diff
@@ -192,6 +193,12 @@ _init(LogPipe *s)
   return grouping_parser_init_method(s);
 }
 
+static int next_clone_id()
+{
+  static guint id = 0;
```

Comment:
I don't think this will work.

This variable is shared between all `grouping-by()` instances, which is not good because the `clone_id` changes if a `grouping-by()` driver gets configured/unconfigured.

E.g.: `A` gets 0, `B` gets 1, `C` gets 2. We remove `A` and `B` from the config, so `C` will get 0 instead of 2.